### PR TITLE
Text resize bug

### PR DESCRIPTION
**Pull request by larrypo**
Addresses issues with adding text or changing font size to a label that is center or right aligned.

### CODE REVIEW

**Comment by islemaster** [comment]
Spot-check looks great too!

I've noticed one "unbalanced" behavior, which I don't think you need to fix in this PR but might be worth running by Poorva as a follow-up fix later.  It's how the box behaves when hitting screen edges left and right aligned:

Left-aligned | Right-aligned
--- | ---
left_aligned | out

**Comment by islemaster** [comment]
The behavior of the center-aligned size-to-fit is fantastic!
out

**Comment by islemaster** [comment]
Looks like there are a few lint errors:
```
[CDO]/apps/src/applab/designElements/label.jsx
  240:14  error  Expected '!==' and instead saw '!='              eqeqeq
  240:32  error  Expected '!==' and instead saw '!='              eqeqeq
  245:9   error  'isBestSize' is assigned a value but never used  no-unused-vars
  245:41  error  Expected '===' and instead saw '=='              eqeqeq
  245:81  error  Expected '===' and instead saw '=='              eqeqeq

✖ 5 problems (5 errors, 0 warnings)
```
You should be able to repro this locally by running `npm run lint` from the apps directory.  If you add our git hooks for your local repo by running `rake install:hooks` from the repository root, you'll also get fast precommit linting of changed files (can be bypassed with the `--no-verify` flag if necessary).

**Comment by islemaster** [comment]
[New build started](https://circleci.com/gh/code-dot-org/code-dot-org/35833).

**Comment by islemaster** [comment]
:tada: Tests passed! :white_check_mark: I'm comfortable merging this fix as-is and following up with new tests in a separate PR, or integrating the tests into this PR.  What would you prefer?

Our builds now upload coverage data to Codecov.io.  You can view coverage for your pull request on the "Files changed" tab by [installing the Codecov browser extension](https://github.com/codecov/browser-extension#choose-your-browser-below).  It looks like your diff is at 44% coverage from existing tests right now.

**Comment by islemaster** [comment]
[New build here.](https://circleci.com/gh/code-dot-org/code-dot-org/35896)

File: apps/src/applab/gridUtils.js [hunk 1]
```diff
@@ -59,6 +59,14 @@ export function isMouseEventInBounds(mouseEvent, container) {
 }
 
 /**
+ * While in design mode, elements get wrapped in a ui-draggable container.
+ * @returns {true} If element is currently wrapped
+ */
+function isDraggableContainer(element) {
```
Comment (islemaster) [inline apps/src/applab/gridUtils.js]: I think you're missing an `export` keyword here.

File: apps/test/unit/applab/designElements/labelTest.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+
+import {expect} from '../../../util/configuredChai';
+
+const label = require('@cdo/apps/applab/designElements/label');
+const library = require('@cdo/apps/applab/designElements/library');
```
Comment (islemaster) [inline apps/test/unit/applab/designElements/labelTest.js]: Could use ES6 import syntax for these, for consistency:
```jsx
import label from '@cdo/apps/applab/designElements/label';
import library from '@cdo/apps/applab/designElements/library';
```

**Comment by islemaster** [comment]
The new unit tests look great!  I just saw the one issue, it looks like the helper you extracted isn't exported.

**Comment by larrypo** [comment]
Fixed and fixed.

**Comment by islemaster** [comment]
[Build in progress.](https://circleci.com/gh/code-dot-org/code-dot-org/35904)  I've got a good feeling about this one :grin:.  I'll merge this PR once it passes.  Thanks @larrypo!

**Comment by islemaster** [comment]
I spoke too soon, there are a couple of lint errors in labelTest.js:
- No need to import `react`, you don't use it.
- We require a newline at the end of the file.

**Comment by islemaster** [comment]
[Here's the new build](https://circleci.com/gh/code-dot-org/code-dot-org/35927).

**Comment by islemaster** [comment]
Tests passed! :tada: I'll merge this tomorrow when I'm in the office.

**Pull request merged**